src/networking.c

```diff
@@ -355,6 +355,15 @@ void _addReplyProtoToList(client *c, const char *s, size_t len) {
 void _addReplyToBufferOrList(client *c, const char *s, size_t len) {
     if (c->flags & CLIENT_CLOSE_AFTER_REPLY) return;
 
+    /* Replicas should normally not cause any writes to the reply buffer. In case a rogue replica sent a command on the
+     * replication link that caused a reply to be generated we'll simply disconnect it. */
+    if (getClientType(c) == CLIENT_TYPE_SLAVE) {
```

Comment:
I think this would more logically fit in `prepareClientToWrite()`, which does similar checks about whether or not a client should actually be written to.

Reply:
one problem is that `feedReplicationBuffer` also calls `prepareClientToWrite`.
so in fact we need to distinguish between `feedReplicationBuffer` and `addReply*` in some way.
i guess the actual way is if it reaches `_addReplyToBufferOrList` or not.

Reply:
Hi @madolson , we also call `prepareClientToWrite` in `prepareReplicasToWrite`

Reply:
That's what I thought. Adding the check in `prepareClientToWrite()` means we've already copied data to the client's output buffer which is what I'm trying to avoid by adding the check to `_addReplyToBufferOrList`. Arguably we can tolerate adding stuff to the client's output buffer and then marking the client to be closed before writing any of it by putting the check in `prepareClientToWrite()`, but I think it's less explicit.

Reply:
Sounds good, is it too much to document that information? It's very obtuse that the only way to check a commands response is that is being added here.

Reply:
i'm only guessing that it's documented in a few places. but i guess adding more can't do any harm.
do you have a specific suggestion?
should we document it here (in _addReplyToBufferOrList)?
i.e. say that `feedReplicationBuffer` doesn't go here, but if addReply* is used on a replica it'll get here.

Reply:
A totally different approach could have been to say something like this:
- Before processing, if we got the command from a replica we validate the command is flagged `ALLOWED_FROM_REPLICA`.
- Commands that are flagged in such a way must adhere to the following:
  - If they want to generate output (error or otherwise) and they're executed from a replica link they close the connection and do nothing.
  - Otherwise they make sure they don't generate any output and do whatever they need (`REPLCONF ACK`).
- If we get a command that's without `ALLOWED_FROM_REPLICA` set, we close the connection.
Now we don't need any special checks in the connection's output buffer write path.

Reply:
i don't like it.. too much work for a small issue.
i prefer the current solution and maybe adding a comment to make it clearer